spec.html

```diff
@@ -26469,7 +26475,7 @@ <h1>Function.prototype.toString ( )</h1>
         <p>When the `toString` method is called, the following steps are taken:</p>
         <emu-alg>
           1. Let _func_ be the *this* value.
-          1. If _func_ is a <emu-xref href="#sec-bound-function-exotic-objects">bound function exotic object</emu-xref> or a <emu-xref href="#sec-built-in-function-objects">built-in function object</emu-xref>, then return an implementation-defined String source code representation of _func_. The representation must have the syntax of a |NativeFunction|. Additionally, if _func_ is a <emu-xref href="#sec-well-known-intrinsic-objects">Well-known Intrinsic Object</emu-xref> and is not identified as an anonymous function, the portion of the returned String that would be matched by |PropertyName| must be the initial value of the *"name"* property of _func_.
+          1. If _func_ is a <emu-xref href="#sec-bound-function-exotic-objects">bound function exotic object</emu-xref> or a <emu-xref href="#sec-built-in-function-objects">built-in function object</emu-xref>, then return an implementation-defined String source code representation of _func_. The representation must have the syntax of a |NativeFunction|. Additionally, if _func_ has an [[InitialName]] internal slot and _func_.[[InitialName]] is a String, the portion of the returned String that would be matched by |NativeFunctionAccessor?| |PropertyName| must be the value of _func_.[[InitialName]].
```

Comment:
This could be changed to:
```suggestion
          1. If _func_ is a <emu-xref href="#sec-built-in-function-objects">built-in function object</emu-xref>, then return an implementation-defined String source code representation of _func_. The representation must have the syntax of a |NativeFunction|. Additionally, if _func_ has an [[InitialName]] internal slot and _func_.[[InitialName]] is a String, the portion of the returned String that would be matched by |NativeFunctionAccessor?| |PropertyName| must be the value of _func_.[[InitialName]].
```

Since [Bound Function exotic objects](https://tc39.es/ecma262/#sec-bound-function-exotic-objects) can be handled by the same step that handles [Proxy exotic objects](https://tc39.es/ecma262/#proxy-exotic-object) and they don’t have `[[InitialName]]`.

Reply:
Should [[InitialName]] be carried over by bound function exotic objects? It seems like that was implied by the "initial name property"

Reply:
I don’t think so given that bound function exotic objects aren’t intrinsics.

Reply:
i think this is correct as is. The text puts the propertyname requirement only on functions which have an [[InitialName]] slot that has a string.

Reply:
That being said the bound functions are all over the place too

```js
'use strict';
const F = Object.getOwnPropertyDescriptor(Map.prototype, 'size').get;
print('1', F.toString());
print('2', F.bind(null).toString());
```

```
┌────────────────┬───────────────────────────────────────────────┐
│ ChakraCore     │ 1 get size                                    │
│                │ 2 function() {                                │
│                │     [native code]                             │
│                │ }                                             │
├────────────────┼───────────────────────────────────────────────┤
│ engine262      │ 1 function get size() { [native code] }       │
│ engine262+     │ 2 function bound get size() { [native code] } │
├────────────────┼───────────────────────────────────────────────┤
│ GraalJS        │ 1 function get size() { [native code] }       │
│                │ 2 function bound() { [native code] }          │
├────────────────┼───────────────────────────────────────────────┤
│ JavaScriptCore │ 1 function get size() {                       │
│                │     [native code]                             │
│                │ }                                             │
│                │ 2 function get size() {                       │
│                │     [native code]                             │
│                │ }                                             │
├────────────────┼───────────────────────────────────────────────┤
│ Moddable XS    │ 1 function get size (){[native code]}         │
│                │ 2 function bound get size (){[native code]}   │
├────────────────┼───────────────────────────────────────────────┤
│ QuickJS        │ 1 function get size() {                       │
│                │     [native code]                             │
│                │ }                                             │
│                │ 2 function bound get size() {                 │
│                │     [native code]                             │
│                │ }                                             │
├────────────────┼───────────────────────────────────────────────┤
│ SpiderMonkey   │ 1 function size() {                           │
│                │     [native code]                             │
│                │ }                                             │
│                │ 2 function() {                                │
│                │     [native code]                             │
│                │ }                                             │
├────────────────┼───────────────────────────────────────────────┤
│ V8             │ 1 function get size() { [native code] }       │
│                │ 2 function () { [native code] }               │
└────────────────┴───────────────────────────────────────────────┘
```

Reply:
My expectation (which we discussed doing in a followon PR) is that all of those with "get" or "set", would basically be `get x() {` or `bound get x() {`.